public/js/index.js

```diff
@@ -3307,11 +3366,45 @@ editorInstance.on('focus', function (editor) {
   personalInfo.cursor = editor.getCursor()
   socket.emit('cursor focus', editor.getCursor())
 })
+editorInstance.on('scroll', function (editor) {
+  SpellChecker.closeOverlay()
+})
+editorInstance.on('refresh', function (editor) {
+  // Close the overlay when the editor is refreshed or resized.
+  // This event is useful for invalidating cached values
+  // that depend on the editor or character size, as mentioned in the documentation.
+  SpellChecker.closeOverlay()
+})
 
 const cursorActivity = _.debounce(cursorActivityInner, cursorActivityDebounce)
 
 function cursorActivityInner (editor) {
   if (editorHasFocus() && !Visibility.hidden()) {
+    const cursorPosition = editor.getCursor()
+    const token = editor.getTokenAt(cursorPosition)
+
+    if (SpellChecker.hasError(token)) {
+      const match = token.state.overlay.match
+      const coords = editor.charCoords(match.position)
+      // Replace the match with the selected suggestion
+      const replaceTextMatch = (value) => {
+        const newValue = value === DELETE_DOUBLE_SPACE_VALUE ? ' ' : value
```

Comment:
Kind of nitpicking but: I admit, from just reading it, this line feels weird.

Would it make more sense if this exception was handled [here](https://github.com/betagouv/pad.numerique.gouv.fr/pull/45/files#diff-995c343281a1957695ad76d4b49344a9dbbbe3a3d03e7bdc6cc16b3272cd80ccR243) instead? And the world outside the SpellChecker code wouldn't have to deal with edge-cases like this one.

Reply:
you totally true, I will improve this in the v2.